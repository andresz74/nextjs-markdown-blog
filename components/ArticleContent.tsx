'use client'; // Mark this component as client-side

import React from 'react';
import Markdown from '@/components/Markdown';
import styles from './ArticleContent.module.css';

interface ArticleContentProps {
    articleContent: string | null;
    loading: boolean;
}

const ArticleContent: React.FC<ArticleContentProps> = ({ articleContent, loading }) => {
    return (
        <main>
            {loading ? (
                <div className={styles.loaderWrapper}>
                    <div className={styles.spinner} data-testid="spinner"></div>
                </div>
            ) : (
                <article>
                    {articleContent ? <Markdown>{articleContent}</Markdown> : <p>Article not found.</p>}
                </article>
            )}
        </main>
    );
};

export default ArticleContent;
