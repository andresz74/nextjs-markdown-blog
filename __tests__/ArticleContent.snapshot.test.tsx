import React from 'react';
import { render } from '@testing-library/react';
import ArticleContent from '@/components/ArticleContent';

describe('ArticleContent snapshot', () => {
  it('matches snapshot for simple markdown', () => {
    const markdown = `# Snapshot Test\n\nThis is **bold** and _italic_.`;
    const { container } = render(
      <ArticleContent articleContent={markdown} loading={false} />
    );

    expect(container).toMatchSnapshot();
  });
});
