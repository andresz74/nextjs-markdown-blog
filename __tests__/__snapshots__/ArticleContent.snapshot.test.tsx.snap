// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleContent snapshot matches snapshot for simple markdown 1`] = `
<div>
  <main>
    <article>
      <h1>
        Snapshot Test
      </h1>
      

      <p>
        This is 
        <strong>
          bold
        </strong>
         and 
        <em>
          italic
        </em>
        .
      </p>
    </article>
  </main>
</div>
`;
